css/grid-layout.css

```diff
@@ -0,0 +1,165 @@
+/*********************** Section 1 ******************/
+
+.navBar {
+    grid-area: navigation;
+    background-color: var(--background-pri-color);
+    color: var(--primary-font-color);
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-end;
+    align-items: center;
+    flex: 1;
+    height: 6rem;
+    position: fixed;
+    right: 0;
+    left: 0;
+    z-index: 2;
+}
+
+.title-Name {
+    margin-right: auto;
+    margin-left: 1rem;
+    font-weight: bold;
+    font-size: 2rem;
+
+}
+
+.navBar-image {
+    max-width: 30%;
+    height: 10%;
+    width: 10%;
+}
+
+.search-container,
+.search-placeholder {
+    background-color: var(--background-sec-color);
+    padding: 0.5rem;
+    margin: 0rem 1rem;
+    border: none;
+    border-radius: 1.5rem;
+}
+
+.search-container {
+    min-width: 16rem;
+    display: flex;
+}
+
+.fa-search {
+    padding-right: 0.5rem;
+    color: var(--primary-font-color);
+}
+
+.navBar>.nav-content {
+    justify-content: space-around;
+    margin: 1rem;
+}
+
+
+/*************** Section 2 ****************/
+.sideMenu {
+    grid-area: sidemenu;
+    background-color: var(--background-pri-color);
+    display: flex;
+    flex-direction: column;
+    position: fixed;
+    top: 5rem;
+    flex: 1;
+    flex-wrap: wrap;
+    height: 100%;
+    overflow-y: scroll;
+    overflow-x: hidden;
+    padding-bottom: 5rem;
+}
+
+.sideMenu-nav-list {
+    list-style-type: none;
+    padding: 1rem;
+    margin: 0rem;
+}
+
+
+.sideMenu li {
+    padding: 0.5rem 1rem;
+    margin: 1rem 0.5rem;
+    font-size: 1.5rem;
+    border-radius: 0rem 1.5rem 1.5rem 0rem;
+}
+
+.sideMenu li:hover {
+    background: linear-gradient(107deg, #f36fff 0%, #4d006b 100%);  
+}
+
+.sideMenu li a:hover {
+    color: var(--background-pri-color);
+    transition: all 0.5s ease-in-out;
+}
+
+
+/************ Section 3 *************/
+.mainContent {
+    grid-area: mainContainer;
+    background-color: var(--background-sec-color);
+    padding: 1rem;
+    border-radius: 1rem 0rem 0rem 1rem;
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+    flex-wrap: wrap;
+    position: relative;
+    top: 6rem;
+    max-width: 100%;
+}
+
+.content-section {
```

Comment:
Hi Arnav!

When I click on a link in the side navigation, the page jumps to content-section, but the header inside this section gets hidden behind the fixed navigation bar at the top.

This might help:
https://stackoverflow.com/questions/4086107/fixed-page-header-overlaps-in-page-anchors

Here's a nice article on the same:
https://css-tricks.com/hash-tag-links-padding/

I had a similar issue. Adding a fixed-height ::before pseudo-element to the section container, whose height is equal to that of the top navbar worked for me.